src/java/org/apache/cassandra/db/compaction/unified/AdaptiveController.java

```diff
@@ -135,19 +145,77 @@ static Controller fromOptions(Environment env,
+            else
+            {
+                logger.warn("Unable to read scaling_parameters. Using starting value instead.");
+                Arrays.fill(scalingParameters, scalingParameter);
+                Arrays.fill(previousScalingParameters, scalingParameter);
+            }
+            if (jsonObject.get("current_flush_size") != null && flushSizeOverrideMB == 0)
+            {
+                currentFlushSize = (long) jsonObject.get("current_flush_size");
+                logger.debug("Successfully read stored current_flush_size from disk");
+            }
+        }
+        catch (IOException e)
+        {
+            logger.warn("Unable to read saved options. Using starting value instead: ", e);
+            Arrays.fill(scalingParameters, scalingParameter);
+            Arrays.fill(previousScalingParameters, scalingParameter);
+        }
+        catch (ParseException e)
+        {
+            logger.warn("Unable to parse stored options. Using starting value instead: ", e);
+            Arrays.fill(scalingParameters, scalingParameter);
+            Arrays.fill(previousScalingParameters, scalingParameter);
+        }
+
         int minScalingParameter = options.containsKey(MIN_SCALING_PARAMETER) ? Integer.parseInt(options.get(MIN_SCALING_PARAMETER)) : DEFAULT_MIN_SCALING_PARAMETER;
         int maxScalingParameter = options.containsKey(MAX_SCALING_PARAMETER) ? Integer.parseInt(options.get(MAX_SCALING_PARAMETER)) : DEFAULT_MAX_SCALING_PARAMETER;
         int intervalSec = options.containsKey(INTERVAL_SEC) ? Integer.parseInt(options.get(INTERVAL_SEC)) : DEFAULT_INTERVAL_SEC;
         double threshold = options.containsKey(THRESHOLD) ? Double.parseDouble(options.get(THRESHOLD)) : DEFAULT_THRESHOLD;
         int minCost = options.containsKey(MIN_COST) ? Integer.parseInt(options.get(MIN_COST)) : DEFAULT_MIN_COST;
         int maxAdaptiveCompactions = options.containsKey(MAX_ADAPTIVE_COMPACTIONS) ? Integer.parseInt(options.get(MAX_ADAPTIVE_COMPACTIONS)) : DEFAULT_MAX_ADAPTIVE_COMPACTIONS;
 
-        return new AdaptiveController(MonotonicClock.preciseTime, env, scalingParameters, previousScalingParameters, survivalFactors, dataSetSizeMB, numShards, minSstableSizeMB, flushSizeOverrideMB, maxSpaceOverhead, maxSSTablesToCompact, expiredSSTableCheckFrequency, ignoreOverlapsInExpirationCheck, l0ShardsEnabled, intervalSec, minScalingParameter, maxScalingParameter, threshold, minCost, maxAdaptiveCompactions);
+        return new AdaptiveController(MonotonicClock.preciseTime, env, scalingParameters, previousScalingParameters, survivalFactors, dataSetSizeMB, numShards, minSstableSizeMB, flushSizeOverrideMB, currentFlushSize, maxSpaceOverhead, maxSSTablesToCompact, expiredSSTableCheckFrequency, ignoreOverlapsInExpirationCheck, l0ShardsEnabled, intervalSec, minScalingParameter, maxScalingParameter, threshold, minCost, maxAdaptiveCompactions);
+    }
+
+    private static int[] validateStoredScalingParameters(JSONArray storedScalingParameters)
```

Comment:
Nit: `readAndValidate...` would be a better description. Or actually just `read...` as it seems I got the wrong impression about what it does.